### ISSUE TITLE
[FEATURE] - Trim option for all data formats

### ISSUE THREAD

**Issue by mhdkassir** [ebiznext/comet-data-pipeline#158]
## Description
Currently, an `Attribute` has a position parameter of type `Position`, and `Position` can have a trim parameter. 
```
case class Attribute(
  ....
  position: Option[Position] = None,
  ...
}
case class Position(first: Int, last: Int, trim: Option[Trim])
```
According to the value of trim, comet will remove spaces from the value of the attribute (to the left, to the right, or on both sides) just before validation. This a very useful option, but with the current model we need to have the position parameter in order to set trim. Meaning we can only use it for data sources with a POSITION format.

This feature proposes  to have the trim parameter at the `Attribute` level which will give a wider possibility to use it even for non-positional data. For example,it is possible to have a DSV file where attributes have extra spaces before the separator or at the end of the line. Using a trim parameter in this case can resolve a validation problem.
_Example:
   attribute1&nbsp;&nbsp;&nbsp;&nbsp;,attribute2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;,12345&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;,attribute3_      
The  third attribute is validated with 5 numeric characters regex, but with the extra spaces the validation will fail. Applying a right trim before validation will resolve this problem.

**This issue exists in many data sources of a client using Comet.**

## Solution proposition
The solution proposed is to define the trim parameter in the `Attribute` case class directly.
case class Attribute(
  ....
  trim: Option[Trim]
  position: Option[Position] = None,
  ...
}
case class Position(first: Int, last: Int)

This can have an impact on some existing ingestion schemas. Existing yml files for Position data sources should be slightly modified. With the generation of yml from excel files, this should not be complicated to do.

## Possible implementation
Update Attribute and Position models, and modify validation code.
Modify XlsReader to comply with new model
Update existing test resources especially domain yml files. 
## Complexity estimation
T-Shirt Size: M

**Comment by hayssams** [comment]
+1

**Comment by mhdkassir** [comment]
implemented in #178
